File: 26-03-2024-Additive-Sequence.py
class Solution:
    def isAdditiveSequence(self, str):
        n = len(str)
        if n < 3:
            return False
        ans, s1 = "", ""
        for i in range(min(n-1, 10)):
            s1 += str[i]
            s2 = ""
            for j in range(i+1, min(n, 10)):
                s2 += str[j]
                ans = s1 + s2
                a = int(s1)
                b = int(s2)
                while len(ans) < n:
                    sum = a + b
                    s = str(sum)
                    if n - len(ans) < len(s):
                        break
                    ans += s
                    a = b
                    b = sum
                    if ans == str:
                        return True
        return False

File: 26-03-2024-Additive-Sequence.cpp
// User function template for C++
class Solution {
  public:
    bool isAdditiveSequence(string str) {
        // Your code here
           int n = str.size();
        if(n<3)
            return 0;
            
        string ans,s1;
        
        for(int i=0;i<min(n-1,10);i++){
            s1.push_back(str[i]);
            string s2 = "";
            for(int j=i+1;j<min(n,10);j++){
                s2.push_back(str[j]);
                ans = s1+s2;
                int a = stoi(s1);
                int b = stoi(s2);
                while(ans.size()<n){
                    int sum = a+b;
                    string s = to_string(sum);
                    if(n-ans.size() < s.size())
                        break;
                    ans += s;
                    a=b;
                    b=sum;
                    if(ans == str)
                        return 1;
                }
            }
        }
        
        return 0;
    }
};
